util/dvsim/utils.py

```diff
@@ -602,7 +602,11 @@ def clean_odirs(odir, max_odirs, ts_format=TS_FORMAT):
     if os.path.exists(odir):
         # If output directory exists, back it up.
         ts = datetime.fromtimestamp(os.stat(odir).st_ctime).strftime(ts_format)
-        shutil.move(odir, odir.with_name(ts))
+        # Prior to Python 3.9, shutil may run into an error when passing in
+        # Path objects (see https://bugs.python.org/issue32689). While this
+        # has been fixed in Python 3.9, string casts are added so that this
+        # also works with older versions.
+        shutil.move(str(odir), str(odir.with_name(ts)))
```

Comment:
It will probably be a bit cleaner to enforce the input `odir` is always `str` and convert it to `Path` variable on line 612 instead.

Reply:
Hm... if we did that, I think we can only get rid of the first cast here. I.e., the second one `str(odir.with_name(ts))` would have to be rewritten as `str(Path(odir).with_name(ts))`, and I am not sure that is much cleaner...

Reply:
Yeah - I am good with cleaning this up later.